geode-management/src/main/java/org/apache/geode/management/runtime/RuntimeInfo.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.runtime;
+
+import java.io.Serializable;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, property = "class")
+public interface RuntimeInfo extends Serializable {
```

Comment:
Could `extends Serializable` be dropped?  Jackson does not require it; however adding it increases the attack surface for Java serialization vulnerability exploits.

Reply:
The implementation of RuntimeInfo (like RuntimeRegionInfo) is gathered on each member and will need to be transferred back to the locator via function calls, so it needs to be Serializable. Geode protects itself from Java serialization vulnerability by allowing only those declared in "sanctioned-serializable.txt" being sent across the wires. If we don't declare it to be Serializable, the function execution will fail.

Reply:
Ok, that makes sense.